pytransform3d/plot_utils/_artists.py

```diff
@@ -299,3 +299,111 @@ def draw(self, renderer):
     def do_3d_projection(self, renderer=None):
         # This supports both matplotlib 3.4 and 3.5
         return 0
+
+
+class Camera(artist.Artist):
+    """A Matplotlib artist that displays a camera.
+
+    This function is inspired by Blender's camera visualization. It will
+    show the camera center, a virtual image plane, and the top of the virtual
+    image plane.
+
+    Parameters
+    ----------
+    M : array-like, shape (3, 3)
+        Intrinsic camera matrix that contains the focal lengths on the diagonal
+        and the center of the the image in the last column. It does not matter
+        whether values are given in meters or pixels as long as the unit is the
+        same as for the sensor size.
+
+    A2B : array-like, shape (4, 4)
+        Transform from frame A to frame B
+
+    virtual_image_distance : float, optional (default: 1)
+        Distance from pinhole to virtual image plane that will be displayed.
+        We assume that this distance is given in meters.
+
+    sensor_size : array-like, shape (2,), optional (default: [1920, 1080])
+        Size of the image sensor: (width, height). It does not matter whether
+        values are given in meters or pixels as long as the unit is the same as
+        for the sensor size.
+
+    kwargs : dict, optional (default: {})
+        Additional arguments for the plotting functions, e.g. alpha.
+    """
+
+    def __init__(
+        self, M, A2B, virtual_image_distance=1.0, sensor_size=(1920, 1080), **kwargs
+    ):
+        super(Camera, self).__init__()
+
+        if "c" in kwargs:
+            color = kwargs.pop("c")
+        elif "color" in kwargs:
+            color = kwargs.pop("color")
+        else:
+            color = "k"
+
+        focal_length = np.mean((M[0, 0], M[1, 1]))
+        sensor_corners = np.array(
```

Comment:
Previously, there were the functions `_make_camera_frame` and `_make_camera_top` in the camera module. I'd recommend to do a similar refactoring here (extract method) to make the code more readable.